metaflow/util.py

```diff
@@ -300,7 +300,7 @@ def dict_to_cli_options(params):
 
                     # Of the value starts with $, assume the caller wants shell variable
                     # expansion to happen, so we pass it as is.
-                    if value.startswith("$"):
+                    if value.lstrip("\\").startswith("$"):
```

Comment:
@romain-intel Can we leave a comment in the code about this change so that a future commit doesn't end up reverting it.